src/main/java/org/spongepowered/common/command/SpongeCommandFactory.java

```diff
@@ -560,9 +561,15 @@ private static CommandSpec createSpongePluginsCommand() {
                 .arguments(optionalWeak(literal(Text.of("reload"), "reload")), optional(plugin(Text.of("plugin"))))
                 .executor((src, args) -> {
                     if (args.hasAny("reload") && src.hasPermission("sponge.command.plugins.reload")) {
-                        src.sendMessage(Text.of("Sending reload event to all plugins. Please wait."));
                         Sponge.getCauseStackManager().pushCause(src);
-                        SpongeImpl.postEvent(SpongeEventFactory.createGameReloadEvent(Sponge.getCauseStackManager().getCurrentCause()));
+                        if (args.hasAny("plugin")) {
+                            PluginContainer plugin = args.<PluginContainer>getOne("plugin").get();
+                            src.sendMessage(Text.of("Sending reload event to " + plugin.getId() + ". Please wait."));
+                            ((SpongeEventManager) Sponge.getEventManager()).post(SpongeEventFactory.createGameReloadEvent(Sponge.getCauseStackManager().getCurrentCause()), plugin);
```

Comment:
You can use `SpongeImpl.getEventManager()` to get the implementation class variant, instead of casting here.

Reply:
I checked for something like that, but `SpongeImpl` doesn't have anything.

Reply:
Maybe write one then? 😉

Reply:
That opens up a lot of questions into this PR, as if something like that is added than the rest of the codebase should be updated as well. Would this just be a method with an `instanceof` check, or should it be caching it the same as the other managers? At that point, this would need to be done by updating Guice injections which is definitely beyond where I want to go here.

Reply:
Then don't, we'll do a rewrite to provide the appropriately casted event manager at some point in the future.